### PR TITLE
Doc update with purpose agreement

### PR DESCRIPTION
**Pull request by jpengar**
#### What type of PR is this?

* documentation

#### What this PR does / why we need it:

As per 04/10 working group call discussions:

| **Issue** | Owner | Description |
| ----------| ----- | ----------- |
| [#32](https://github.com/camaraproject/IdentityAndConsentManagement/issues/32) ([PR#64](https://github.com/camaraproject/IdentityAndConsentManagement/pull/64)) | Orange | **The concept of purpose**<br> - Robert Ludovic (Orange) asks if we can write this in GSMA. Telefonica indicates that it would be ok to move the content finally included to GSMA, either to the playbook or to the relevant chapter in GitHub.<br>- Jesús (Telefonica) says that a new PR can be created to fix issue #59 (updating the flows with the final agreement) and to include the agreement in the CAMARA documentation. And the corresponding flows in the GSMA github repository could be updated accordingly.<br>It is proposed to close issue #32 when this is done, and to continue the discussion on medium-term alternatives in a separate issue, starting a "clean" discussion, as this issue is already too long on github. |

This PR is intended to fix the issue #59 correcting the flows according to agreed purpose solution and to fix the #32 documenting the purpose solution agreed by the active CAMARA participant on this issue.

#### Which issue(s) this PR fixes:

It fixes issues #59 and #32

#### Special notes for reviewers:

This PR document reached agreement on purpose solution for the short-term, discussions will be continued for the mid-term in a new issue as per 04/10 working group call discussions.

### CODE REVIEW

**Review by bigludo7 — APPROVED** [review]
LGTM Thanks

**Comment by Elisabeth-Ericsson** [comment]
The proprietary format proposed for the scope claim (concatenating purpose from dpvw3c with data scope), documented in issue #32  cannot be detected by an AuthZ server, unless it is prefixed by a NAME SPACE indicator e.g. opg#.
AuthZ servers of CSPs generate access tokens for various APIs exposed by CSPs, not only Camara APIs. Such they need to know when they have to implement proprietary logic.

File: documentation/CAMARA-API-access-and-user-consent.md [hunk 1]
```diff
@@ -71,7 +71,40 @@ A purpose declares what the application intends to do with a set of personal inf
...
+  - one **API name** to cover the complete API (This API name is not described in the API spec YAML file as technical scope) – In this case the request covers all technical scopes of the API.
+
+<br>
+
+The examples are provided below:
+
+Strong recommendation for format samples:
+
+```
+fraud-preventiondetection:retrieve-sim-swap-date
```
[the remark this answers is not in the document]
Comment (bigludo7) [reply documentation/CAMARA-API-access-and-user-consent.md:90]: @eric-murray good point ! 
@jpengar we should avoid on mapping there. Let's use the w3c  purpose term (`FraudPreventionAndDetection`)?

[the remark this answers is not in the document]
Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md:90]: @eric-murray good catch!
@bigludo7 Yes, I actually took the examples as they were in the github issue comment, and I didn't realise at the time that the dpv value format was actually not the one specified by the W3C Data Privacy Vocabulary. We should use the values as defined by W3C: https://w3c.github.io/dpv/dpv/#vocab-purpose
I can correct this.

[the remark this answers is not in the document]
Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md]: This is fixed by commit https://github.com/camaraproject/IdentityAndConsentManagement/pull/75/commits/e7eae0ab6d62b0a68302a5f0845ebc7a40052a53

File: documentation/CAMARA-API-access-and-user-consent.md [hunk 2]
```diff
@@ -71,7 +71,40 @@ A purpose declares what the application intends to do with a set of personal inf
 
 ### Applying purpose concept in the authorization request
 
->[TO BE EDITED/COMPLETED] - [OAuth 2.0 Authorization Framework](https://www.rfc-editor.org/rfc/rfc6749)
+In order to declare a purpose when accessing CAMARA APIs, it is strongly recommended that `scope` param in the /authorize request will be valued with:
```
Comment (eric-murray) [inline documentation/CAMARA-API-access-and-user-consent.md:74]: > in the /authorize request

Is this proposal only intended to apply to the front-end flow? CIBA flows and client credentials require scopes as well! Surely this proposal applies to all authorisation schemes.

Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md:74]: Yes, it applies to all authorisation schemes. Let me rephrase that a bit, as I tried to keep the original github comment to avoid any misunderstanding. But you've raised a fair point. May be just removing that /authorize part is enough... you can also see that, for example, the auth code flow and the CIBA flow are already updated with this agreement.

File: documentation/CAMARA-API-access-and-user-consent.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shilpa-padgaonkar) [reply documentation/CAMARA-API-access-and-user-consent.md]: yes it was a typo error from my side in the issue comment

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by rartych — APPROVED** [review]
LGTM

Remarks for further updates: 

- According to OAuth-2 spec:

`The value of the scope parameter is expressed as a list of space-delimited, case-sensitive strings. `
then we need to clarify if multiple scopes with proposed syntax are possible.

- In CIBA specs we have: 

`CIBA authentication requests MUST therefore contain the openid scope value.`
 Probably OpenID Foundation can guide us if `openid` scope value should be included.
